src/frontend/app/utils/apiRequest.ts

```diff
@@ -0,0 +1,126 @@
+import superagent from 'superagent'
+// import { appInfo, versionNumber } from '../config'
+import { validJson, wordsEqual, regHttp, handleUserCookies } from './utils'
+// import { resultHandler } from './resultHandler'
+import { resultHandler } from './errorHandler'
+import { handleLogout } from './executeActions'
+
+const _ = require('./underscore')
+const methods = ['get', 'post', 'put', 'patch', 'del']
+
+function formatUrl (path: string) {
+  const isHttpPath = regHttp.test
+  const hasSlash = (path[0] === '/')
+  const adjustedPath = (isHttpPath || hasSlash) ? path : '/' + path
+  const apiPrefix = '' // api
+  return apiPrefix + adjustedPath;
+}
+
+function jsonToUrlParams (data: any) {
+  if (typeof data !== 'object') return ''
+  return Object.keys(data).map(k => {
+      return encodeURIComponent(k) + '=' + encodeURIComponent(data[k])
+  }).join('&');
+}
+
+export default class ApiRequest {
+  constructor() {
+
+    let ths: Record<string, any> = {}
+    ths = this
+
+    methods.forEach(method =>
+
+      ths[method] = (
+        path: any,
+        {
+          params,
+          data,
+          requestAction
+        }: {
+          params: any,
+          data: any,
+          requestAction: any
+        }
+      ) => new Promise((resolve, reject) => {
```

Comment:
`reject` is not used here, how do you handle API errors or network errors?

Reply:
reject method has been temporarily removed by me, normally should be 
`err ? reject : resolve`
There's a resultHandler method imported in this file, that's the place to handle errors before reject